### PR TITLE
Documentation pod status reporting

### PR DESCRIPTION
**Pull request by jpodivin**

### CODE REVIEW

File: docs/deploying.md [hunk 1]
```diff
@@ -481,6 +481,25 @@ OpenStackDataPlaneNodeSet has the following status fields:
...
+##### OpenStackDataPlaneNodeSet Pods
+
+Pods within OpenStackDataPlaneNodeSet can fail for many reasons.
+Under normal circumstances, these failures may be transient.
+However, for facilitating triage, the operator does check on service pods and reports
+if their status is indicative of failure.
+
+Failure of individual pods within nodeset is logged on info level in format:
+
+ `"openstack dataplane pod <POD_NAME> failed due to <FAILURE_REASON> message: <FAILURE_MSG>"`
```
Comment (slagle) [inline docs/deploying.md:493]: I think we should change this log message in the code as well, for all the above reasons.

File: docs/deploying.md [hunk 2]
```diff
@@ -481,6 +481,25 @@ OpenStackDataPlaneNodeSet has the following status fields:
 | DNSClusterAddresses | |
 | CtlplaneSearchDomain | |
 
+##### OpenStackDataPlaneNodeSet Pods
+
+Pods within OpenStackDataPlaneNodeSet can fail for many reasons.
```
Comment (slagle) [inline docs/deploying.md:486]: There aren't pods in a nodeset, and I don't think that's the right way to think about the design conceptually. The pods this section are talking about are actually started by a dataplane Deployment by way of AnsibleEE by way of a Job, then a pod.

File: docs/deploying.md [hunk 3]
```diff
@@ -481,6 +481,25 @@ OpenStackDataPlaneNodeSet has the following status fields:
...
+Under normal circumstances, these failures may be transient.
```
Comment (slagle) [inline docs/deploying.md:487]: I don't understand this. Transient failures aren't normal circumstances. There shouldn't be any failures. The environment might cause transient or inconsistent failures (due to networking perhaps), but that's not normal.

File: docs/deploying.md [hunk 4]
```diff
@@ -481,6 +481,25 @@ OpenStackDataPlaneNodeSet has the following status fields:
...
+However, for facilitating triage, the operator does check on service pods and reports
+if their status is indicative of failure.
```
Comment (slagle) [inline docs/deploying.md:489]: By service pod, do you mean the pod started by the job by the AnsibleEE by the Deployment for a DataPlaneService execution? This isn't clear.

**Review by slagle — CHANGES_REQUESTED** [review]

File: docs/deploying.md (hunk 2 above)
Comment (jpodivin) [reply docs/deploying.md:486]: Good point.

File: docs/deploying.md (hunk 3 above)
Comment (jpodivin) [reply docs/deploying.md:487]: Shouldn't be yes. I'll drop that.

File: docs/deploying.md (hunk 1 above)
Comment (jpodivin) [reply docs/deploying.md:493]: What kind of message would you prefer?

File: docs/deploying.md (hunk 4 above)
Comment (jpodivin) [reply docs/deploying.md:489]: Right, the multiple levels of delegation are hard to keep track of, I better define it at the start.

File: docs/deploying.md (hunk 1 above)
Comment (jpodivin) [reply docs/deploying.md]: Message changed.

File: docs/assemblies/deploying.adoc [hunk 5]
```diff
@@ -688,3 +688,26 @@ $ oc get openstackdataplanenodeset openstack-edpm-ipam -o json | jq '.status.con
   "type": "NodeSetDNSDataReady"
 }
 ----
+
+=== Dataplane deployment Pods
+
+Pods launched during deployement throught OpenStackAnsiblEEE managed jobs,
+further referred to as pods, are are marked with label `osdpd=<OpenStackDataPlaneDeployment.Name>`
+and checked for failures.
```
[the remark this answers is not in the document]
Comment (jpodivin) [reply docs/assemblies/deploying.adoc:696]: We really ought to mention the `ospd=` label though.

**Comment by jpodivin** [comment]
recheck

**Comment by jpodivin** [comment]
recheck

File: docs/assemblies/deploying.adoc [hunk 6]
```diff
@@ -688,3 +688,34 @@ $ oc get openstackdataplanenodeset openstack-edpm-ipam -o json | jq '.status.con
...
+
+
+During reconciliation of OpenStackDataPlaneDeployment resources,
+OpenStackAnsibleEE resources, marked with label `osdpd=<OpenStackDataPlaneDeployment.Name>,
+are created to manage the ansible executions that configure the EDPM nodes.
+The OpenStackAnsibleEE resources create Kubernetes Jobs resources, which in turn create Kubernetes Pods that are executing ansible.
+
+Checking the logs of these Jobs and Pods can show ansible failures and help diagnose issues.
+
+Pods launched during deployement throught OpenStackAnsiblEEE managed jobs,
```
Comment (slagle) [inline docs/assemblies/deploying.adoc:702]: "The pods launched..."

*through

File: docs/assemblies/deploying.adoc [hunk 7]
```diff
@@ -688,3 +688,34 @@ $ oc get openstackdataplanenodeset openstack-edpm-ipam -o json | jq '.status.con
...
+further referred to as pods, are are marked with label `osdpd=<OpenStackDataPlaneDeployment.Name>`
```
Comment (slagle) [inline docs/assemblies/deploying.adoc:703]: drop "further referrred to as pods" clause

File: docs/assemblies/deploying.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (slagle) [reply docs/assemblies/deploying.adoc:704]: :+1:

File: docs/assemblies/deploying.adoc [hunk 8]
```diff
@@ -688,3 +688,34 @@ $ oc get openstackdataplanenodeset openstack-edpm-ipam -o json | jq '.status.con
...
+and checked for failures.
+
+While the recovery of jobs is an automated process,
+additional information can be useful for diagnosing the issue.
+
+Failure of individual pods is logged on info level in format:
+
+[,console]
+----
+ `"openstack job <POD_NAME> failed due to <FAILURE_REASON> with message: <FAILURE_MSG>"`
```
Comment (slagle) [inline docs/assemblies/deploying.adoc:713]: need to update to match the message from the code

**Review by slagle — CHANGES_REQUESTED** [review]

File: docs/assemblies/deploying.adoc (hunk 8 above)
Comment (jpodivin) [reply docs/assemblies/deploying.adoc]: Because of recent documentation changes the doc doesn't exist anymore.

Comment (jpodivin) [reply docs/assemblies/deploying.adoc]: But I've changed the message anyway.

File: docs/assemblies/deploying.adoc (hunk 6 above)
Comment (jpodivin) [reply docs/assemblies/deploying.adoc]: Document doesn't exist anymore.

File: docs/assemblies/deploying.adoc (hunk 7 above)
Comment (jpodivin) [reply docs/assemblies/deploying.adoc]: ditto

**Review by slagle — APPROVED** [review]

**Review by slagle — APPROVED** [review]

**Pull request merged**